### ISSUE TITLE
examples in slot `slope gradient` don't validate

### ISSUE THREAD

**Issue by mslarae13** [microbiomedata/nmdc-schema#1353]
See here 
![Screenshot 2023-11-14 at 3 54 17 PM](https://github.com/microbiomedata/nmdc-schema/assets/44581429/f4588d76-03e1-4a12-b98e-bd3ab8daf4aa)

slot slope_gradient has 10% as a valid example but it won't validate with out a ` ` .. so 10 %

**Comment by turbomam** [comment]
good catch. we are inheriting this from an old version of MIxS. I will check v6.2 to see if the examples that violate our expectations are still there.

if they aren't, then I need to get my act together and update the import process to use v6.2, which will require lots of review form tohers

alternatively we could do either or both of these:
- ask GSC to use our recommended examples
- change the examples *in the submission-schema* to our preferred style

**Comment by mslarae13** [comment]
Pending your inspection of 'if examples that violate our expectations are still there' ...

_If MIxS still has invalid examples_
I recommend we update the validator to not REQUIRE a 'space' between the # and % .. so 10% WILL validate
We put in an issue to MIxS asking for an update to their validator

_If MIxS has removed the invalid examples_
We should pull that in, still make the suggestion to MIxS to allow for 'no space' between # and % but not change it on our end. Wait for MIxS to update.

@turbomam thoughts?

**Comment by turbomam** [comment]
I forgot to look last week. I'll do it now.

**Comment by turbomam** [comment]
I wonder where those examples came from? I don't see them in MIxS 6.0-6.2

## Transposed rows from MIxS 6.1 "mixs_v6.xlsx"

| Environmental package   | soil                                                                                                                                                                                                                    | agriculture                                                                                                                                                                                                             |
|-------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Structured comment name | slope_gradient                                                                                                                                                                                                          | slope_gradient                                                                                                                                                                                                          |
| Package item            | slope gradient                                                                                                                                                                                                          | slope gradient                                                                                                                                                                                                          |
| Definition              | Commonly called 'slope'. The angle between ground surface and a horizontal line (in percent). This is the direction that overland water would flow. This measure is usually taken with a hand level meter or clinometer | Commonly called 'slope'. The angle between ground surface and a horizontal line (in percent). This is the direction that overland water would flow. This measure is usually taken with a hand level meter or clinometer |
| Expected value          | measurement value                                                                                                                                                                                                       | measurement value                                                                                                                                                                                                       |
| Value syntax            | {float} {unit}                                                                                                                                                                                                          | {float} {unit}                                                                                                                                                                                                          |
| Example                 |                                                                                                                                                                                                                         |                                                                                                                                                                                                                         |
| Requirement             | X                                                                                                                                                                                                                       | C                                                                                                                                                                                                                       |
| Preferred unit          | percentage                                                                                                                                                                                                              | percentage                                                                                                                                                                                                              |
| Occurrence              | 1                                                                                                                                                                                                                       | 1                                                                                                                                                                                                                       |
| MIXS ID                 | MIXS:0000646                                                                                                                                                                                                            | MIXS:0000646                                                                                                                                                                                                            |

## MIxS v6.2.0 documentation page
- https://genomicsstandardsconsortium.github.io/mixs/0000646/

## linkml source for the slot, v6.2.0:

```yaml
name: slope_gradient
annotations:
  Preferred_unit:
    tag: Preferred_unit
    value: percentage
description: Commonly called 'slope'. The angle between ground surface and a horizontal
  line (in percent). This is the direction that overland water would flow. This measure
  is usually taken with a hand level meter or clinometer
title: slope gradient
from_schema: https://w3id.org/mixs
keywords:
- slope
slot_uri: MIXS:0000646
alias: slope_gradient
domain_of:
- Agriculture
- Soil
range: string
pattern: ^[-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?( *- *[-+]?[0-9]*\.?[0-9]+(?:[eE][-+]?[0-9]+)?)?
  *([^\s-]{1,2}|[^\s-]+.+[^\s-]+)$
structured_pattern:
  syntax: ^{scientific_float}( *- *{scientific_float})? *{text}$
  interpolated: true
  partial_match: true
```

**Comment by turbomam** [comment]
## Regexr sandbox
- https://regexr.com/7nn8d

**Comment by turbomam** [comment]
If you look at the `modifications_long.tsv` section in the [submission-schema search results for `slope_gradient`](https://github.com/search?q=repo%3Amicrobiomedata%2Fsubmission-schema%20slope_gradient&type=code) you'll see how we have asserted our own examples and a `todo`:

> Slope is a percent. How does the validation work? Check to correct examples

**Comment by turbomam** [comment]
`submission-schema` has a validation `pattern` of `^[-+]?[0-9]*\.?[0-9]+ +\S.*$` for `slope_gradient`, but it isn't coming from MIxS 6.2.0 or from NMDC's frozen MIxS v6.0 either. See  https://microbiomedata.github.io/mixs/slope_gradient/

As far as I am concerned we can assert whatever examples we want in the `Biosample` slot usage in nmdc.yaml and whatever validation pattern we want in https://github.com/microbiomedata/submission-schema/blob/main/sheets_and_friends/tsv_in/modifications_long.tsv

@pkalita-lbl and @mslarae13 this is the kind of thing we will have to keep an eye on when we switch to importing MIxS terms directly from https://github.com/GenomicsStandardsConsortium/mixs/blob/v6.2.0/src/mixs/schema/mixs.yaml. See 

- https://github.com/microbiomedata/nmdc-schema/pull/1386

**Comment by turbomam** [comment]
@mslarae13 I have emphasize a minimal number of abstracted `structured_pattern`s  in MIxS v6.2.0. If you want to suggest changing GSC's pattern for `slope_gradient`, let's to look for the other slots that are using the same pattern and be consistent.

**Comment by mslarae13** [comment]
Thanks @turbomam forgot we added those! Regardless, I think it's the same solution. It's weird to allow "10 %" and not "10%".
